cmd/snap/cmd_run_test.go

```diff
@@ -1820,6 +1802,57 @@ func (s *RunSuite) TestWaitWhileInhibitedTextFlow(c *check.C) {
 	c.Check(meter.Labels, check.DeepEquals, []string{"please wait..."})
 }
 
+func (s *RunSuite) TestWaitWhileInhibitedGraphicalSessionFlow(c *check.C) {
+	restoreIsGraphicalSession := snaprun.MockIsGraphicalSession(true)
+	defer restoreIsGraphicalSession()
+
+	var notification *usersessionclient.PendingSnapRefreshInfo
+	restorePendingRefreshNotification := snaprun.MockPendingRefreshNotification(func(refreshInfo *usersessionclient.PendingSnapRefreshInfo) error {
+		notification = refreshInfo
+		return nil
+	})
+	defer restorePendingRefreshNotification()
+
+	var called int
+	restore := snaprun.MockIsLocked(func(snapName string) (runinhibit.Hint, error) {
+		called++
+		if called < 2 {
+			return runinhibit.HintInhibitedForRefresh, nil
+		}
+		return runinhibit.HintNotInhibited, nil
+	})
+	defer restore()
+
+	c.Assert(runinhibit.LockWithHint("some-snap", runinhibit.HintInhibitedForRefresh), check.IsNil)
+	c.Assert(snaprun.WaitWhileInhibited("some-snap"), check.IsNil)
+	c.Check(called, check.Equals, 2)
+
+	c.Check(s.Stdout(), check.Equals, "")
+	c.Assert(notification, check.NotNil)
```

Comment:
(nitpick) this is not strictly needed, the DeepEquals in the next line should be able to deal with Nil

Reply:
Indeed, thanks.